lib/nodes/opendss.cpp

```diff
@@ -0,0 +1,378 @@
+
+  size_t i;
+  json_t *json_data;
+  json_error_t err;
+
+  json_array_foreach(json, i, json_data) {
+    if (in) {
+      int ret;
+      const char *name;
+      const char *type;
+      json_t *a_mode = nullptr;
+      Element ele;
+      ret = json_unpack_ex(json_data, &err, 0, "{ s: s, s: s, s: o }",
+                          "name", &name, "type", &type, "data", &a_mode);
+      if (ret)
+        throw ConfigError(json, err, "node-config-node-opendss");
+
+      if (!json_is_array(a_mode) && a_mode)
+        throw ConfigError(a_mode, "node-config-opendssIn",
+                          "datatype must be configured as a list of name objects");
+
+      ele.name = name;
+
+      if (!strcmp(type, "generator"))
+        ele.type = ElementType::generator;
+      else if (!strcmp(type, "load"))
+        ele.type = ElementType::load;
+      else if (!strcmp(type, "isource"))
+        ele.type = ElementType::isource;
+      else
+        throw SystemError("Invalid element type '{}'", type);
+
+      size_t n;
+      json_t *json_mode;
+      json_array_foreach(a_mode, n, json_mode) {
+        const char *mode = json_string_value(json_mode);
+        // Assign mode according to the OpenDSS function mode
+        switch (ele.type) {
+          case ElementType::generator:
+            if (!strcmp(mode, "kV"))
+              ele.mode.push_back(1);
+            else if (!strcmp(mode, "kW"))
+              ele.mode.push_back(3);
+            else if (!strcmp(mode, "kVar"))
+              ele.mode.push_back(5);
+            else if (!strcmp(mode, "Pf"))
+              ele.mode.push_back(7);
+            else
+              throw SystemError("Invalid data type: {}", mode);
+            break;
+          case ElementType::load:
+            if (!strcmp(mode, "kW"))
+              ele.mode.push_back(1);
+            else if (!strcmp(mode, "kV"))
+              ele.mode.push_back(3);
+            else if (!strcmp(mode, "kVar"))
+              ele.mode.push_back(5);
+            else if (!strcmp(mode, "Pf"))
+              ele.mode.push_back(7);
+            else
+              throw SystemError("Invalid data type: {}", mode);
+            break;
+          case ElementType::isource:
+            if (!strcmp(mode, "Amps"))
+              ele.mode.push_back(1);
+            else if (!strcmp(mode, "AngleDeg"))
+              ele.mode.push_back(3);
+            else if (!strcmp(mode, "Frequency"))
+              ele.mode.push_back(5);
+            else
+              throw SystemError("Invalid data type: {}", mode);
+            break;
+          default:
+            throw SystemError("Invalid element type");
+            break;
+        }
+      }
+      dataIn.push_back(ele);
+    } else {
+      monitor_name.push_back(json_string_value(json_data));
+    }
+  }
+}
+
+int OpenDSS::parse(json_t *json) {
+
+  int ret = Node::parse(json);
+  if (ret)
+    return ret;
+
+  json_error_t err;
+  json_t *json_format = nullptr;
```

Comment:
The `format` setting appears to be unused? Please remove also from the OpenAPI spec and example configs.